File: CloudMusic/src/main/java/com/ken/android/CloudMusic/Utils/HttpUtils.java
package com.ken.android.CloudMusic.Utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import com.squareup.okhttp.Call;
import com.squareup.okhttp.Callback;
import com.squareup.okhttp.OkHttpClient;
import com.squareup.okhttp.Request;
import com.squareup.okhttp.Response;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;

/**
 * Created by axnshy on 16/5/25.
 */
public class HttpUtils {
    /*
    * 判断网络是否正常
    * */

    public static boolean isServerConnection(Context context){


        ConnectivityManager manager= (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo info=manager.getActiveNetworkInfo();
        if (info==null){
            System.out.println("Null");
            return false;

        }
        else {
            return true;
        }
    }
    /*
    * 使用get的方式访问网络并获取到Json数据
    *
    * */
    public static String doGet(String uriString){
        StringBuffer sb=new StringBuffer();
        String result;
        try {
            URL url=new URL(uriString);
            HttpURLConnection conn= (HttpURLConnection) url.openConnection();

            conn.setConnectTimeout(10*1000);

            conn.setReadTimeout(10*1000);

            InputStream is=conn.getInputStream();

            BufferedReader br=new BufferedReader(new InputStreamReader(is));

            String line=br.readLine();
            System.out.println("line:"+line);

            if (line!=null){

                sb.append(line);
            }
        } catch (MalformedURLException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }

        result=sb.toString();
        /*if(sb.toString().contains("weather_callback")){
            result=sb.toString().substring(sb.toString().indexOf("(")+1,sb.toString().indexOf(")"));
        }*/
        return result;
    }

}

File: SmartPlay/src/main/java/com/luffy/smartplay/utils/HttpUtils.kt
package com.luffy.smartplay.utils

import android.net.ConnectivityManager
import android.content.*
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.MalformedURLException
import java.net.URL

/**
 * Created by axnshy on 16/5/25.
 */
object HttpUtils {
    /*
    * 判断网络是否正常
    * */
    fun isServerConnection(context: Context): Boolean {
        val manager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val info = manager.activeNetworkInfo
        return if (info == null) {
            println("Null")
            false
        } else {
            true
        }
    }

    /*
    * 使用get的方式访问网络并获取到Json数据
    *
    * */
    fun doGet(uriString: String?): String {
        val sb = StringBuffer()
        val result: String
        try {
            val url = URL(uriString)
            val conn = url.openConnection() as HttpURLConnection
            conn.connectTimeout = 10 * 1000
            conn.readTimeout = 10 * 1000
            val `is` = conn.inputStream
            val br = BufferedReader(InputStreamReader(`is`))
            val line = br.readLine()
            println("line:$line")
            if (line != null) {
                sb.append(line)
            }
        } catch (e: MalformedURLException) {
            e.printStackTrace()
        } catch (e: IOException) {
            e.printStackTrace()
        }
        result = sb.toString()
        /*if(sb.toString().contains("weather_callback")){
            result=sb.toString().substring(sb.toString().indexOf("(")+1,sb.toString().indexOf(")"));
        }*/return result
    }
}